packages/grid/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -128,7 +128,28 @@ export const GridRootStyles = styled('div', {
     '--unstable_DataGrid-overlayBackground': theme.vars
       ? `rgba(${theme.vars.palette.background.defaultChannel} / ${theme.vars.palette.action.disabledOpacity})`
       : alpha(theme.palette.background.default, theme.palette.action.disabledOpacity),
+    '--unstable_DataGrid-containerBackground': theme.vars
+      ? `rgba(${theme.vars.palette.background.defaultChannel} / 0.8)`
+      : alpha(theme.palette.background.default, 0.8),
```

Comment:
I think that this is clearly too transparent, preventing to read the header's content with enough contrast. 

This feels better:
```suggestion
      : alpha(theme.palette.background.default, 0.95),
```

Before
![Screenshot 2023-11-27 at 21 27 55](https://github.com/mui/mui-x/assets/3165635/76c6b9ba-1e41-4895-893f-5e7026029b6e)

After
![Screenshot 2023-11-27 at 21 27 45](https://github.com/mui/mui-x/assets/3165635/0c4a6e50-11db-4187-be4b-990036f72d99)

But I'm not sure, no transparency at all might be for the best. I have tried to play with `backdrop-filter: blur(xpx);` as well but I can't find anything that feel better than having no transparency.

Reply:
Yeah this is just testing values so I can better see what's going on. I haven't finalized the colors, I was planning on making it opaque to preserve the design as it is, but some transparency can make it look a bit more modern.

Reply:
Argument against transparency: we need to display more rows on screen and it will affect performance a bit.